### ISSUE TITLE
Issues generating previews in FreeBSD

### ISSUE THREAD

**Issue by DarkRider1768** [cowai/camerarawpreviews#25]
Hey,

Just saw a fixed issue regarding generating previews on FreeBSD. Unfortunately, we are current (literally just built the install yesterday) and we are still having issues with previews. We followed the ticket instructions, perl seemingly generates the preview just fine but it doesn't show in the gallery even with a new upload of the file with a different name, but can't seem to get it to work beyond this.

Any assistance in this would be much appreciated! We are hoping to get this into our clients environment on Friday but told them that there will be some bugs to work out and final configuration to be made. Anything I can do to help you along with this process, I'm more than willing. Just let me know what you need!

**Comment by cowai** [comment]
*cloud log would be great.
If you give me a test environment by ssh or a vm file I can debug locally and try to get it to work. I probably do not have time until Thursday at best.

**Comment by DarkRider1768** [comment]
I modified the command a bit (didn't like the path to exiftool) and it
worked but it didn't seem to work as provided. I'll get you the output when
I get in in the morning.

The directory I used that seemed to let it work was
/usr/local/bin/exiftool... If I remember correctly. I'll get that for you
too (even though it's probably less useful lol).

Thanks for taking a look at this, it's a huge help!

**Comment by cowai** [comment]
My app uses a vendored exiftool and not the system exiftool.
you need to test with the proper path ("/nextcloud/apps/camerarawpreviews/vendor/jmoati/exiftool-bin/exiftool").
If that gives you an error, then we know there is a problem with the vendored exiftool on freebsd.

**Comment by DarkRider1768** [comment]
> root@NextCloud:~ # perl $APP_PATH/vendor/jmoati/exiftool-bin/exiftool -json -preview:all $RAWFILE_PATH
APP_PATH: Undefined variable.
root@NextCloud:~ # perl /usr/local/www/nextcloud/apps/camerarawpreviews/vendor/jmoati/exiftool-bin/exiftool -json -preview:all $RAWFILE_PATH
RAWFILE_PATH: Undefined variable.
root@NextCloud:~ # perl /usr/local/www/nextcloud/apps/camerarawpreviews/vendor/jmoati/exiftool-bin/exiftool -json -preview:all /mnt/data/__groupfolders/1/Flute_AF580-BO_Full\ Front_Raw_216.CR2
[{
  "SourceFile": "/mnt/data/__groupfolders/1/Flute_AF580-BO_Full Front_Raw_216.CR2",
  "PreviewImage": "(Binary data 848238 bytes, use -b option to extract)",
  "ThumbnailImage": "(Binary data 5060 bytes, use -b option to extract)"
}]

This is what I get from command line. No preview showing in NextCloud though. I even uploaded the same file with a different name, as suggested in the older ticket, but it doesn't show anything. I can get you some new logs if you'd like.

**Comment by cowai** [comment]
hmm. I think I would need to debug this locally.
The output from the command seems good. I think maybe perl detection in my app does not work properly with your system.

If you could provide me with a vps or a vm where I could debug it I could help you.
I have very limited time for non-paid work these days to setup a freebsd debug environment so if you would like to speed things up you could put a bounty on this on bountysource or provide me with a server/vm I could run ;)

**Comment by DarkRider1768** [comment]
I'll look into doing an rsync to my home NAS and changing the passwords on stuff. After that, I should be able to get you ssh into it. It's just not going to fly with giving external ssh into a client box at work. After I have a test environment propped up at home, you can hammer at it as much as you'd like. I'm not sure that the client is going to want to pay for the bounty, unfortunately. They've been living off a trial "loaner" NAS from us for about 8 months while they dragged their feet over getting their own.

**Comment by cowai** [comment]
Please try latest version now, v0.6. If you're lucky this is the same bug as https://github.com/cowai/camerarawpreviews/issues/26 which is fixed now.

**Comment by cowai** [comment]
@DarkRider1768 Care to update and check now with v0.6.1 ?

**Comment by andmor-** [comment]
I am seeing the same issue with v0.6.1 on FreeBSD v11.2.0.4, Nextcloud 14.0.3

`perl $APP_PATH/vendor/jmoati/exiftool-bin/exiftool -json -preview:all $RAW_DIR/IMG_20181013_192303.dng`
returns:
`[{
  "SourceFile": "$RAW_DIR/IMG_20181013_192303.dng"
}]`
No errors. But Nextcloud's UI does not show any preview and all requests for .DNG previews end up with 404.

**Comment by cowai** [comment]
Actually that output shows there is no preview inside the file. At least not one exiftool can detect.

// Ari Selseng


---- On Mon, 12 Nov 2018 13:56:12 +0100 notifications@github.com wrote ----



I am seeing the same issue with v0.6.1 on FreeBSD v11.2.0.4, Nextcloud 14.0.3

perl $APP_PATH/vendor/jmoati/exiftool-bin/exiftool -json -preview:all $RAW_DIR/IMG_20181013_192303.dng
 returns:
[{ "SourceFile": "$RAW_DIR/IMG_20181013_192303.dng" }]
 No errors. But Nextcloud's UI does not show any preview and all requests for .DNG previews end up with 404.

**Comment by andmor-** [comment]
OK thank you for your answer. I tried a DNG file from a different device and it works.
My misunderstanding was: this app does not generate the previews if they are not embedded in the raw file in the first place. It only extracts (and adapts) existing previews from raw files and makes them available to Nextcloud.

So it is working great on FreeBSD. but unfortunately can't handle most of my DNG files.

**Comment by cowai** [comment]
Yeah, actually rendering previews from RAW without extract embedded previews requires a lot more power from the server and is very hard getting right colors and levels, its just not worth it.

If you have Adobe Lightroom or equivelent you can normally set it to make previews automatically.
